Common/src/drivers/CDriverBase.cpp

```diff
@@ -0,0 +1,1393 @@
+  }
+
+  return tags;
+}
+
+vector<string> CDriverBase::GetDeformableMarkerTags() const {
+  vector<string> tags;
+  const auto nMarker = main_config->GetnMarker_Deform_Mesh();
+
+  for (auto iMarker = 0u; iMarker < nMarker; iMarker++) {
+    tags.push_back(main_config->GetMarker_Deform_Mesh_TagBound(iMarker));
+  }
+
+  return tags;
+}
+
+unsigned long CDriverBase::GetNumberDimensions() const { return main_geometry->GetnDim(); }
+
+unsigned long CDriverBase::GetNumberElements() const { return main_geometry->GetnElem(); }
+
+unsigned long CDriverBase::GetNumberMarkerElements(unsigned short iMarker) const {
+  if (iMarker >= GetNumberMarkers()) {
+    SU2_MPI::Error("Marker index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return main_geometry->GetnElem_Bound(iMarker);
+}
+
+vector<unsigned long> CDriverBase::GetElementIDs() const {
+  const auto nElem = GetNumberElements();
+
+  vector<unsigned long> values;
+
+  for (auto iElem = 0ul; iElem < nElem; iElem++) {
+    values.push_back(GetElementIDs(iElem));
+  }
+
+  return values;
+}
+
+unsigned long CDriverBase::GetElementIDs(unsigned long iElem) const {
+  if (iElem >= GetNumberElements()) {
+    SU2_MPI::Error("Element index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return main_geometry->elem[iElem]->GetGlobalIndex();
+}
+
+vector<unsigned long> CDriverBase::GetMarkerElementIDs(unsigned short iMarker) const {
+  const auto nBound = GetNumberMarkerElements(iMarker);
+
+  vector<unsigned long> values;
+
+  for (auto iBound = 0ul; iBound < nBound; iBound++) {
+    values.push_back(GetMarkerElementIDs(iMarker, iBound));
+  }
+
+  return values;
+}
+
+unsigned long CDriverBase::GetMarkerElementIDs(unsigned short iMarker, unsigned long iBound) const {
+  if (iBound >= GetNumberMarkerElements(iMarker)) {
+    SU2_MPI::Error("Marker element index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return main_geometry->bound[iMarker][iBound]->GetGlobalIndex();
+}
+
+vector<vector<unsigned long>> CDriverBase::GetElementConnectivities() const {
+  const auto nElem = GetNumberElements();
+
+  vector<vector<unsigned long>> values;
+
+  for (auto iElem = 0ul; iElem < nElem; iElem++) {
+    values.push_back(GetElementConnectivities(iElem));
+  }
+
+  return values;
+}
+
+vector<unsigned long> CDriverBase::GetElementConnectivities(unsigned long iElem) const {
+  if (iElem >= GetNumberElements()) {
+    SU2_MPI::Error("Element index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  unsigned short nNode = main_geometry->elem[iElem]->GetnNodes();
+
+  vector<unsigned long> values;
+
+  for (auto iNode = 0u; iNode < nNode; iNode++) {
+    unsigned long iPoint = main_geometry->elem[iElem]->GetNode(iNode);
+
+    values.push_back(main_geometry->nodes->GetGlobalIndex(iPoint));
+  }
+
+  return values;
+}
+
+vector<vector<unsigned long>> CDriverBase::GetMarkerElementConnectivities(unsigned short iMarker) const {
+  const auto nBound = GetNumberMarkerElements(iMarker);
+
+  vector<vector<unsigned long>> values;
```

Comment:
Is it really worth having this type of function in the interface? This would be one line of python and could be stored in a numpy array which would probably be more efficient than the list of lists than vector<vector> maps to?